### PR TITLE
Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by HansZizold**
### INTEGRATION SPECS AND n+1 PROBLEMS

- Bullet gem was installed according to the procedures and no error appeared in the app. All the features of the app were tested and the rails server was restarted. Logs of the bullet gem installation and the bullet log file are included.

image
image
image

- Integration specs were written according to the instructions. All tests are working.
- This project was developed in pair programming. Participants: @rebaz36 , @HansZizold.

[Complete Requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

Happy review!!!

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -54,6 +54,12 @@ group :development, :test do
   gem 'ffi'
   gem 'rails-controller-testing'
   gem 'rspec-rails'
+
+  # Capybara, the library that allows us to interact with the browser using Ruby
+  gem 'capybara'
+
+  # bullet gem for N+1 problems
+  gem 'bullet'
```
Comment (AshakaE) [inline Gemfile:62]: As a requirement, you're not allowed to use the bullet gem. 
- [ ] Kindly remove the bullet gem and solve the n +1 issue using a different approach

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @HansZizold & @rebaz36 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- pull request ✅ 
- git flow ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HansZizold** [comment]
Dear @AshakaE, 

You're right about the observation. The bullet gem was removed.

We also checked the logs of the post controller - index view.

For the part of the posts there is only one query to the database:

image

For the part of the comments it can be seen one query for the comment part:

image

Best regards,

File: Gemfile (hunk 1 above)
Comment (HansZizold) [reply Gemfile:62]: @AshakaE, OK!

**Review by brytebee — APPROVED** [review]
Hi, @HansZizold👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**